integration-tests/deployment/ccip/add_chain.go

```diff
@@ -193,3 +166,105 @@ func NewChainInboundProposal(
 		"0s", // TODO: Should be parameterized.
 	)
 }
+
+func NewSetCandidateProposal(
+	state CCIPOnChainState,
+	e deployment.Environment,
+	homeChainSel, feedChainSel, newChainSel uint64,
+	tokenConfig TokenConfig,
+	rmnHomeAddress []byte,
+) (*timelock.MCMSWithTimelockProposal, *CreateDonArgs, error) {
+	// Home chain new don.
+	// - Add new DONs for destination to home chain
+	nodes, err := deployment.NodeInfo(e.NodeIDs, e.Offchain)
```

Comment:
@connorwstein  What do you think about moving nodes into `deployment.Environment` ? We end up deriving it in mainly places and it can be formed easily from generic environment